vignettes/case-study-scaling.Rmd

```diff
@@ -0,0 +1,222 @@
+
+```
+shinycannon recording.log https://rsc.example.com/content/1234/ --workers 1 --loaded-duration-minutes 5 --output-dir run1
+```
+
+- `recording.log` is the recording to use 
+- `https://rsc.example.com/content/1234/` is the "solo" URL of the target application
+- `--workers 1` means to simulate one user
+- `--loaded-duration-minutes 5` means each user should continually "loop" through the recording for at least 5 minutes after all workers have started. Since our recording is about a minute long, for this run we'll collect metrics for about 5 iterations of the recording.
+- `--output-dir` is the directory to populate with metrics and run metadata.
+
+> Note: The command to run shinycannon might be different on your platform. For alternatives, see the [shinycannon instructions](https://rstudio.github.io/shinyloadtest/index.html#shinycannon)
+
+Once `shinycannon` has started, you should see output like the following:
+
+```
+2018-08-31 00:22:11.009 INFO [progress] - Running: 0, Failed: 0, Done: 0
+2018-08-31 00:22:11.010 INFO [thread00] - Waiting for warmup to complete
+2018-08-31 00:22:11.011 INFO [thread01] - Warming up
+2018-08-31 00:22:11.011 INFO [thread00] - Maintaining for 5 minutes (300000 ms)
+2018-08-31 00:22:16.010 INFO [progress] - Running: 1, Failed: 0, Done: 0
+2018-08-31 00:22:21.011 INFO [progress] - Running: 1, Failed: 0, Done: 0
+```
+
+- Messages labeled `[progress]` are status messages printed every 5 seconds until the run is complete.
+- Messages labeled `[thread00]` are produced by the parent thread supervising the run, and contain information about state transitions. A run moves through 3 states as it progresses: warming up, maintaining, and cooling down. The warmup and cooldown periods are those during which workers are gradually starting or stopping. Maintenance is the period during which all workers are running, and its length is governed by the `--loaded-duration-minutes` option.
+
+> Note: If you see any messages of severity ERROR, the run hasn't necessarily failed, but the metrics it produces may not be usable. You should investigate the error and re-run. Every shinycannon output directory contains a `debug.log` file which may be useful to look at as you investigate.
+
+### Comparison run
+
+Now that we have baseline metrics, we should conduct a more ambitious run for contrast. Perhaps in our organization we can expect up to 5 people to use cranwhales at the same time, so to simulate that possibility we'll run `shinycannon` again with `--workers 5`:
+
+```
+shinycannon recording.log https://rsc.example.com/content/1234/ --workers 1 --loaded-duration-minutes 5 --output-dir run5
+```
+
+## Analyze
+
+At this point we have two runs worth of metrics in our `run1` and `run5` output directories. Let's get back in an R console and process the data into a dataframe:
+
+```
+library(shinyloadtest)
+df <- load_runs(`Run 1` = "run1", `Run 5` = "run5")
```

Comment:
idk where it started and i'm in a tossup as to if it should be used.

`load_runs(`Run 1` = "run1", `Run 5` = "run5")` is equivalent to `load_runs("Run 1" = "run1", "Run 5" = "run5")`

back-ticks are not very standard. Thoughts?

Reply:
`"` work for me